core/src/main/java/com/amplifyframework/util/UserAgent.java

```diff
@@ -20,26 +20,83 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 
+import com.amplifyframework.AmplifyException;
+import com.amplifyframework.Platform;
+import com.amplifyframework.core.Amplify;
 import com.amplifyframework.core.BuildConfig;
+import com.amplifyframework.logging.Logger;
+
+import java.util.Map;
 
 /**
  * A utility to construct a User-Agent header, to be sent with all network operations.
  */
 public final class UserAgent {
+    private static final Logger LOG = Amplify.Logging.forNamespace("amplify:core");
+    private static final int SIZE_LIMIT = 254; // VARCHAR(254)
+
     private static String instance = null;
 
     private UserAgent() {}
 
+    /**
+     * Configure User-Agent singleton with Amplify configuration instance.
+     * @param platformVersions A map of additional platforms that are calling this library
+     *                         and their respective versions.
+     * @throws AmplifyException If called twice or user-agent exceeds size limit.
+     */
+    public static synchronized void configure(@NonNull Map<Platform, String> platformVersions)
```

Comment:
If you move `Platform` to be `AmplifyConifguration.Platform`, then do something like:

```
LinkedHashMap<String, String> paltformVersions.= new LinkedHashMap<>();
for (Map.Entry<Platform, String> provided : inputVersions.entrySet()) {
    platformVersion.put(provided.getKey().name(), provided.getValue());
}
```
Then you could break the type dependency and just have String-to-String map on this API.

Reply:
Moving to `UserAgent` instead